### PR TITLE
Ignore complex props

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/webstudio-is/webstudio/discussions/37

There are props like object and function we cannot express with UI. Even more there is style prop which if casted to string gives runtime error. So the best way would be to ignore such types.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - test it on preview
- [ ] hi @istarkov, I need you to do
  - detailed review (read every line)

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts [hunk 1]
```diff
@@ -2,10 +2,8 @@ import type { PropMeta } from "@webstudio-is/generate-arg-types";
 
 export const props: Record<string, PropMeta> = {
   slot: { required: false, control: "text", type: "string" },
-  style: { required: false, control: "text", type: "string" },
   title: { required: false, control: "text", type: "string" },
   defaultChecked: { required: false, control: "boolean", type: "boolean" },
-  defaultValue: { required: false, control: "text", type: "string" },
```
Comment (kof) [inline packages/sdk-components-react/src/__generated__/vimeo.props.ts:8]: defaultValue shouldn't be removed?

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts [hunk 2]
```diff
@@ -19,7 +17,6 @@ export const props: Record<string, PropMeta> = {
   accessKey: { required: false, control: "text", type: "string" },
   autoFocus: { required: false, control: "boolean", type: "boolean" },
   className: { required: false, control: "text", type: "string" },
-  contentEditable: { required: false, control: "text", type: "string" },
```
Comment (kof) [inline packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: not really necessary but isn't a complex type, so I think this indicates mistake in the logic

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts [hunk 3]
```diff
@@ -41,7 +38,6 @@ export const props: Record<string, PropMeta> = {
   about: { required: false, control: "text", type: "string" },
   content: { required: false, control: "text", type: "string" },
   datatype: { required: false, control: "text", type: "string" },
-  inlist: { required: false, control: "text", type: "string" },
```
Comment (kof) [inline packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: also seems type string

**Review by kof — APPROVED** [review]
check the things that this change got removed from props, seems like some of them are not complex object types, its also strange that style type doesn't work as a string, because you can totally set it as a string `document.body.style = ''color: red'`

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 2 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: It's "Booleanish" type

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 1 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:8]: Seems like should be defined by component as html one is very generic.

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 2 above)
Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: can we detect such types from the data?

**Comment by TrySound** [comment]
Not strange. We generate not dom but react props.

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 1 above)
Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:8]: Yeah, I wonder if our inference is inaccurate or the types are actually inaccurate from the source

Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:8]: Not sure why defaultValue is defined on all components even div
```
        defaultValue?: string | number | ReadonlyArray<string> | undefined;
```

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 3 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: any in types

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 2 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: The question is which type we should prefer
```
boolean | 'true' | 'false'
```

Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: form that choice this is clearly a boolean

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 3 above)
Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: I can't even find an attribute inlist anywhere in the spec, how did we get there, how can we get rid of it and maybe some similar cases?

**Comment by kof** [comment]
> Not strange. We generate not dom but react props.

Right, we basically have a fundamental mismatch, because not all attribute names map exactly 1:1 to react props. This is a fundamental problem and given our cross-framework direction for the future we need to think how we can make sure the html native attributes always map correctly to whatever framework is doing

**Comment by kof** [comment]
Another such missmatch is "class". We need a way to solve this for all frameworks, any ideas?

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 2 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:22]: Here yes but
```
'aria-checked'?: boolean | 'false' | 'mixed' | 'true' | undefined;
'aria-current'?: boolean | 'false' | 'true' | 'page' | 'step' | 'location' | 'date' | 'time' | undefined;
```

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 3 above)
Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: React.HTMLAttributes

**Comment by TrySound** [comment]
There is no universal way if we allow user components. Component types is the only source of props we can have.

**Comment by kof** [comment]
Right, but in our case we extending the html attributes in the type, maybe this is a bad approach since not all html attributes can be passed from react to the dom

File: packages/sdk-components-react/src/__generated__/vimeo.props.ts (hunk 3 above)
Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: super weird, I can't find it in the spec at all

Comment (kof) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: we are already using ComponentProps<'div'> so we are probably already relying on React.HTMLAttributes

Comment (TrySound) [reply packages/sdk-components-react/src/__generated__/vimeo.props.ts:44]: Yes, this is what's inside of React.HTMLAttributes

**Comment by TrySound** [comment]
1. we do not pass from react to dom
2. HTMLAttributes is React specific type, it matches perfectly
3. The issue was in casting everything including style object to text control

So, no casting - no issue

**Review by kof — APPROVED** [review]

**Pull request merged**